File: Challenge1/java/src/main/java/us/airraisin/dailyprogrammer/PromptName.java
package us.airraisin.dailyprogrammer;

import java.io.IOException;
import java.text.MessageFormat;
import java.util.Scanner;

public class PromptName 
{
    public static void main( String[] args )
    {
    	String name = null, userName = null;
    	int age = 0;
    
    	try (Scanner scanner = new Scanner(System.in)) { 
	    	while (name == null) {
		    	System.out.println("What is your name? ");
				name = scanner.next();
	    	}
	    	
	    	while (age <= 0) {
	    		System.out.println("What is your age? ");
	    		age = scanner.nextInt();
	    	}
	    	
	    	while (userName == null) {
	        	byte [] buff = new byte [1024];
		    	System.out.println("What is your Reddit username? ");
				userName = scanner.next();
	    	}
    	}
    	
    	System.out.println(MessageFormat.format("your name is {0}, you are {1} years old, and your username is {2}", name, age, userName));
    }
}

File: Challenge1/python/python/PromptName.py
'''
Created on Dec 27, 2019

@author: rick
'''

if __name__ == '__main__':
    name = input("What is your name: ");
    age = input("What is your age: ");
    userName = input("What is your Reddit username: ");
    outputString = "your name is {}, you are {} years old, and your username is {}\n"
    outputString = outputString.format(name, age, userName)
    print(outputString);
    logfile = open("logfile.txt", "a")
    logfile.write(outputString);
    logfile.close()